src/packages/recompose/withReducer.js

```diff
@@ -7,11 +7,18 @@ const withReducer = (stateName, dispatchName, reducer, initialState) =>
     const factory = createEagerFactory(BaseComponent)
     return class extends Component {
       state = {
-        stateValue: typeof initialState === 'function'
-          ? initialState(this.props)
-          : initialState
+        stateValue: this.initalizeStateValue()
       };
 
+      initalizeStateValue() {
+        if (typeof initialState !== 'undefined') {
```

Comment:
Don't know.
This is just my habit from the time when it was possible to define variable named `undefined`